lib/Middleware/SessionMiddleware.php

```diff
@@ -48,108 +55,185 @@ public function beforeController(Controller $controller, string $methodName): vo
 			return;
 		}
 
-		$reflectionMethod = new \ReflectionMethod($controller, $methodName);
+		//ASSERTION
+		$documentId = $this->getDocumentId();
+		$this->document = $this->documentService->getDocument($documentId);
```

Comment:
This always runs - even if non of the `Require...` attributes were set.

Before we'd only get the document if any of the attributes was there.
Is this intentional?

I don't think it matters in practice as I assume all `ISessionAwareController`s require at least one of them. Just want to point out a change in behavior that i am seeing here.